applications/jonny5/src/j5_balance_crawler.erl

```diff
@@ -0,0 +1,229 @@
+-export([delayed_hangup/2]).
+
+-include("jonny5.hrl").
+
+-record(state, {}).
+-define(SERVER, ?MODULE).
+-define(CRAWLER_CYCLE, 60000).
+-define(INTERACCOUNT_DELAY, 10).
+
+%%====================================================================
+%% API
+%%====================================================================
+%%--------------------------------------------------------------------
+%% Function: start_link() -> {ok,Pid} | ignore | {error,Error}
+%% Description: Starts the server
+%%--------------------------------------------------------------------
+start_link() ->
+  gen_server:start_link(?MODULE, [], []).
+
+%%====================================================================
+%% gen_server callbacks
+%%====================================================================
+
+%%--------------------------------------------------------------------
+%% Function: init(Args) -> {ok, State} |
+%%                         {ok, State, Timeout} |
+%%                         ignore               |
+%%                         {stop, Reason}
+%% Description: Initiates the server
+%%--------------------------------------------------------------------
+init([]) ->
+    wh_util:put_callid(?MODULE),
+    self() ! 'crawl_accounts',
+    {'ok', #state{}}.
+
+%%--------------------------------------------------------------------
+%% Function: %% handle_call(Request, From, State) -> {reply, Reply, State} |
+%%                                      {reply, Reply, State, Timeout} |
+%%                                      {noreply, State} |
+%%                                      {noreply, State, Timeout} |
+%%                                      {stop, Reason, Reply, State} |
+%%                                      {stop, Reason, State}
+%% Description: Handling call messages
+%%--------------------------------------------------------------------
+handle_call(_Request, _From, State) ->
+    Reply = {'error', 'not_implemented'},
+    {'reply', Reply, State}.
+
+%%--------------------------------------------------------------------
+%% Function: handle_cast(Msg, State) -> {noreply, State} |
+%%                                      {noreply, State, Timeout} |
+%%                                      {stop, Reason, State}
+%% Description: Handling cast messages
+%%--------------------------------------------------------------------
+handle_cast(_Msg, State) ->
+    {'noreply', State}.
+
+%%--------------------------------------------------------------------
+%% Function: handle_info(Info, State) -> {noreply, State} |
+%%                                       {noreply, State, Timeout} |
+%%                                       {stop, Reason, State}
+%% Description: Handling all non call/cast messages
+%%--------------------------------------------------------------------
+handle_info('crawl_accounts', _State) ->
+    IsEnabled = whapps_config:get_is_true(?APP_NAME, <<"balance_crawler_enabled">>, true),
+    case j5_channels:accounts() of
```

Comment:
You could do:

`erlang
case IsEnabled andalso j5_channels:accounts() of
    'false' -> next_cycle();
    [] -> next_cycle();
    Accounts -> crawl(Accounts)
end
```

Avoiding the work of `j5_channels` if not enabled.

In fact, thinking about this more, the IsEnabled check should be in `init/1` or even `start_link/0`, returning `'ignore'` if not enabled.